### ISSUE TITLE
mutable-class-default (RUF012) misinterprete the code block

### ISSUE THREAD

**Issue by ArazHeydarov** [astral-sh/ruff#9647]
Hello, 

RUF012 working principle seems to be broken. I'm defining a Pydantic [json_encoder](https://docs.pydantic.dev/1.10/usage/exporting_models/#json_encoders) config however Ruff detects it as typing annotation and raises an error when I check. 

Code:
```Python
from bson import ObjectId
from pydantic import BaseConfig, BaseModel


class MongoModel(BaseModel):
    class Config(BaseConfig):
        allow_population_by_field_name = True
        json_encoders = {ObjectId: str}
```

Command: `ruff check .`

Result:
```
src/models/mongo_model.py:8:25: RUF012 Mutable class attributes should be annotated with `typing.ClassVar`
Found 1 error.
```

Current settings:
```toml
[tool.ruff.lint]
extend-select = ["I", "C90", 'N', 'UP', 'ASYNC', 'RUF']
```  

Current Ruff version:
`ruff 0.1.14`

**Comment by charliermarsh** [comment]
It looks like `BaseConfig` is missing from our allowlist.

**Comment by charliermarsh** [comment]
What is `BaseConfig`? I can't find it in the Pydantic docs.

**Comment by trag1c** [comment]
> What is `BaseConfig`? I can't find it in the Pydantic docs.

It's part of Pydantic v1: https://docs.pydantic.dev/2.0/api/config/#pydantic.config.BaseConfig

**Comment by ArazHeydarov** [comment]
It's a class that is deprecated as of Pydantic 2.0. The result is the same without inheriting from that class.
I know that the Config class definition itself is deprecated.

**Comment by charliermarsh** [comment]
Thank you! Will add.
